### PR TITLE
WIP: Scheduler simulator

### PR DESCRIPTION
**Pull request by wyegelwel**

### CODE REVIEW

**Comment by wyegelwel** [comment]
I need to add a test confirming that the example output provided is the same as what is produced by the simulator.

File: scheduler/test/cook/test/simulator.clj [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+            (org.joda.time.DateTimeUtils/setCurrentMillisFixed (inc (.getTime (tc/to-date (t/now)))))
+            (async/>!! rebalancer-trigger-chan rebalancer-complete-chan)
+            (async/<!! rebalancer-complete-chan)
+            (log/info "Rebalance complete")
+
+            ;; Request for jobs that are complete to have cook be notified
+            (log/info "Send completion status to scheduler")
+            (org.joda.time.DateTimeUtils/setCurrentMillisFixed (inc (.getTime (tc/to-date (t/now)))))
+            (async/>!! complete-trigger-chan :trigger)
+            (async/>!! complete-trigger-chan :trigger)
```
Comment (shamsimam) [inline scheduler/test/cook/test/simulator.clj]: Please add a comment that we are relying on the processing of this second trigger to be a no-op.

**Comment by wyegelwel** [comment]
@shamsimam Thanks for the feedback, splitting the PR into those pieces is a great suggestion. I will do that now.

File: scheduler/src/cook/components.clj [hunk 2]
```diff
@@ -83,41 +85,47 @@
...
+                              mesos-gpu-enabled
+                              rebalancer
+                              {:fenzo-max-jobs-considered fenzo-max-jobs-considered
+                               :fenzo-scaleback fenzo-scaleback
+                               :fenzo-floor-iterations-before-warn fenzo-floor-iterations-before-warn
+                               :fenzo-floor-iterations-before-reset fenzo-floor-iterations-before-reset
+                               :fenzo-fitness-calculator fenzo-fitness-calculator
+                               :good-enough-fitness good-enough-fitness}
+                              ; TODO Make these configurable..
+                              {:ranker-trigger-chan (chime-ch  (periodic/periodic-seq (t/now) (t/seconds 5)))
```
Comment (wyegelwel) [inline scheduler/src/cook/components.clj]: WHAT?!?!

**Pull request closed without merging**